addons/hw_drivers/iot_handlers/drivers/L10nKeEDISerialDriver.py

```diff
@@ -149,8 +159,14 @@ def send(self, msgs):
                 request = struct.pack('B%ds2sB' % (len(core_message)), STX, core_message, self.generate_checksum(core_message), ETX)
                 time.sleep(self._protocol.commandDelay)
                 self._connection.write(request)
-                time.sleep(self._protocol.measureDelay)
-                response = self._connection.read_all()
+                # If we know the expected output size, we can set the read
+                # buffer to match the size of the output.
+                output_size = COMMAND_OUTPUT_SIZE.get(msg[0])
+                if output_size:
+                    response = self._connection.read(COMMAND_OUTPUT_SIZE.get(msg[0]))
```

Comment:
Is there a timeout for this?  (should we catch the exception?)

Reply:
there is a timeout, it's defined within the protocol (the timeout is defined on the serial object)